universal-application-tool-0.0.1/app/modules/SecurityModule.java

```diff
@@ -70,17 +98,22 @@ protected FormClient provideFormClient() {
 
   @Provides
   @Singleton
-  protected Config provideConfig(
-      DirectBasicAuthClient directBasicAuthClient, FormClient formClient) {
+  protected ProfileFactory provideProfileFactory() {
+    return new ProfileFactory();
+  }
+
+  @Provides
+  @Singleton
+  protected Config provideConfig(GuestClient guestClient, FormClient formClient) {
     // This must match the line in `routes` also.
     Clients clients = new Clients(baseUrl + "/callback");
-    clients.setClients(directBasicAuthClient, formClient);
+    clients.setClients(guestClient, formClient);
     PlayHttpActionAdapter.INSTANCE
         .getResults()
         .putAll(
             ImmutableMap.of(
                 HttpConstants.UNAUTHORIZED,
-                unauthorized("401 not authorized").as(HttpConstants.HTML_CONTENT_TYPE),
+                redirect("/loginForm?message=login"),
```

Comment:
Suggestion (blocking): use the reverse router when feasible: `controllers.routes.HomeController.loginForm().url() + "?message=login"`